benchexec/tools/witnesslint.py

```diff
@@ -25,10 +26,17 @@ def version(self, executable):
         version_string = self._version_from_tool(executable)
         return version_string.partition("version")[2].strip().split(" ")[0]
 
+    def _code_matches_tool_output(self, tool_output, exit_code):
+        pattern = re.compile(rf"Exit code: {exit_code}")
+        if pattern.match(tool_output[-1]):
+            return True
+        return False
+
     def determine_result(self, run):
         if run.exit_code.value == 0:
-            return result.RESULT_TRUE_PROP
+            if self._code_matches_tool_output(run.output, 0):
```

Comment:
It is up to you to decide, but is it worth double checking here? (I find it redundant and surprising for a tool to output its own exit code.)

In general, I would expect that a tool either outputs its result as some kind of meaningful message (not `Exit code: 0`), and/or sets its exit code appropriately. The tool-info module could then check either, or both, of course.

Reply:
I am not 100% sure that this is necessary here, but in the next line it is, to prevent misinterpretations of the exit code.
For example, in this table (https://sv-comp.sosy-lab.org/2021/results/results-verified/cpa-seq.results.SV-COMP21.All.table.html#/table - last column) we get a wrong result: an exception occurs and we get 1 as exit code - but not because this was set by the linter, so it should not be reported as `false`.

The additional check is to make sure that the perceived exit code was indeed set by witnesslint.